src/tests/functional/shared_test_classes/src/base/utils/generate_static_shapes.cpp

```diff
@@ -0,0 +1,151 @@
+
+uint64_t clip(uint64_t n, uint64_t lower, uint64_t upper) {
+    return std::max(lower, std::min(n, upper));
+}
+
+void clip_restrict_dims(InputShape& input_shape, const std::map<size_t, size_t>& restrict_dims) {
+    std::vector<ov::Shape>& staticShapes = input_shape.second;
+    for (const auto& pair : restrict_dims) {
+        uint64_t dimMax = std::numeric_limits<char>::max();
+        uint64_t dimMin = pair.second;
+        auto& dim0 = staticShapes[0][pair.first];
+        auto& dim2 = staticShapes[2][pair.first];
+        if (dim0 != dim2) {
+            dim0 = clip(dim0, dimMin, dimMax);
+            dim2 = clip(dim2, dimMin, dimMax);
+        }
+    }
+}
+
+namespace {
+
+InputShape generate(const std::shared_ptr<ov::Node>& node,
+                    const std::shared_ptr<ov::op::v0::Parameter>& param) {
+    std::vector<ov::Shape> staticShapes = { param->get_partial_shape().get_min_shape(),
+                                            param->get_partial_shape().get_min_shape(),
+                                            param->get_partial_shape().get_max_shape() };
+    ov::Shape midShape;
+    for (const auto s : param->get_partial_shape()) {
+        int dimValue = 1;
+        if (s.is_dynamic()) {
+            size_t range = s.get_max_length() - s.get_min_length();
+            if (range > std::numeric_limits<char>::max()) {
+                ov::test::utils::fill_data_random(&range, 1, std::numeric_limits<char>::max(), s.get_min_length(), 1);
+            }
+            ov::test::utils::fill_data_random(&dimValue, 1, range, s.get_min_length(), 1);
+        } else {
+            dimValue = s.get_length();
+        }
+        midShape.push_back(dimValue);
+    }
+    staticShapes[1] = midShape;
+
+    // Shape validation to avoid large values
+    uint64_t dimMin = 1;
+    uint64_t dimMax = std::numeric_limits<char>::max();
+    for (int i = 0; i < staticShapes[0].size(); ++i) {
+        auto& dim0 = staticShapes[0][i];
+        auto& dim2 = staticShapes[2][i];
+        if (dim0 != dim2) {
+            dim0 = clip(dim0, dimMin, dimMax);
+            dim2 = clip(dim2, dimMin, dimMax);
+        }
+    }
+    return InputShape{param->get_partial_shape(), staticShapes};
+}
+
+InputShape generate(const std::shared_ptr<ov::op::v1::Convolution>& node,
+                    const std::shared_ptr<ov::op::v0::Parameter>& param) {
+    std::map<size_t, size_t> restrict_dims;
+    for (size_t i = 0; i < param->get_output_size(); i++) {
+        for (const auto &node : param->get_output_target_inputs(i)) {
+            std::shared_ptr<ov::Node> nodePtr = node.get_node()->shared_from_this();
+            for (size_t port = 0; port < nodePtr->get_input_size(); ++port) {
+                if (nodePtr->get_input_node_ptr(port)->shared_from_this() != param) {
+                    auto constant_ptr = std::dynamic_pointer_cast<ov::op::v0::Constant>(nodePtr->get_input_node_shared_ptr(port));
+                    if (constant_ptr == nullptr) {
```

Comment:
Potentially, some convolution inputs can be constant.. Looks like using `input_port_id` can help to avoid this cycle and potential issue with constant ports